Maple2.Server.Game/PacketHandlers/MapleopolyHandler.cs

```diff
@@ -0,0 +1,191 @@
+                return;
+            }
+        }
+
+        // roll two dice
+        int dice1 = Random.Shared.Next(1, 7);
+        int dice2 = Random.Shared.Next(1, 7);
+
+        totalTileValue += dice1 + dice2;
+        if (dice1 == dice2) {
+            freeRollValue++;
+        }
+
+        session.GameEventUserValue[GameEventUserValueType.MapleopolyFreeRollAmount] = freeRollValue;
+        session.GameEventUserValue[GameEventUserValueType.MapleopolyTotalTileCount] = totalTileValue;
+        session.Send(MapleopolyPacket.Roll(totalTileValue, dice1, dice2));
+    }
+
+    private void HandleResult(GameSession session) {
+        using GameStorage.Request db = GameStorage.Context();
+        GameEvent? gameEvent = db.FindEvent(nameof(BlueMarble));
+        if (gameEvent?.EventInfo is not BlueMarble blueMarble) {
+            // TODO: Find error to state event is not active
+            return;
+        }
+
+        int.TryParse(session.GameEventUserValue.Get(GameEventUserValueType.MapleopolyFreeRollAmount, gameEvent).Value, out int freeRollValue);
+        int.TryParse(session.GameEventUserValue.Get(GameEventUserValueType.MapleopolyTotalTileCount, gameEvent).Value, out int totalTileValue);
+
+        int currentTilePosition = totalTileValue % blueMarble.Tiles.Count;
+
+        BlueMarbleTile tile = blueMarble.Tiles[currentTilePosition];
+
+        switch (tile.Type) {
+            case BlueMarbleTileType.Item:
+            case BlueMarbleTileType.TreasureTrove:
+                if (!ItemMetadata.TryGet(tile.Item.ItemId, out ItemMetadata? itemMetadata)) {
+                    // TODO: Error
+                    break;
+                }
+                var item = new Item(itemMetadata, tile.Item.ItemRarity, tile.Item.ItemAmount);
+                if (!session.Item.Inventory.Add(item, true)) {
+                    // TODO: Mail to user
+                }
+                break;
+            case BlueMarbleTileType.Backtrack:
+                totalTileValue -= tile.MoveAmount;
+                break;
+            case BlueMarbleTileType.MoveForward:
+                totalTileValue += tile.MoveAmount;
+                break;
+            case BlueMarbleTileType.RoundTrip:
+                totalTileValue += blueMarble.Tiles.Count;
+                break;
+            case BlueMarbleTileType.GoToStart:
+                totalTileValue += blueMarble.Tiles.Count - currentTilePosition;
+                break;
+            case BlueMarbleTileType.Start:
+                // Do nothing
+                break;
+            case BlueMarbleTileType.Lose:
+            case BlueMarbleTileType.RollAgain:
+            case BlueMarbleTileType.Trap:
+            default:
+                Logger.Warning($"Unhandled Mapleopoly tile type: {tile.Type}");
+                break;
+        }
+
+        GameEventUserValue totalTrips = session.GameEventUserValue.Get(GameEventUserValueType.MapleopolyTotalTrips, gameEvent);
```

Comment:
It might help to split this logic into a helper function:

Helper:
```cs
private void RewardNewTrips(GameSession session, GameEvent gameEvent, int totalNewTrips) {
    GameEventUserValue totalTrips = session.GameEventUserValue.Get(GameEventUserValueType.MapleopolyTotalTrips, gameEvent);
    int.TryParse(totalTrips.Value, out int trips);
    while (trips < totalNewTrips) {
        ...
    }
}
```

Usage:
```cs
int totalNewTrips = totalTileValue / blueMarble.Tiles.Count;
RewardNewTrips(session, gameEvent, totalNewTrips);
session.GameEventUserValue[GameEventUserValueType.MapleopolyTotalTrips] = totalNewTrips;
...
```